gossip/service/gossip_service.go

```diff
@@ -199,6 +202,28 @@ func (jcm *joinChannelMessage) AnchorPeersOf(org api.OrgIdentityType) []api.Anch
 	return jcm.members2AnchorPeers[string(org)]
 }
 
+// anchorPeerTracker maintains anchor peer endpoints for all the channels.
+type anchorPeerTracker struct {
+	// allEndpoints contains anchor peer endpoints for all the channels,
+	// its key is channel name, value is map of anchor peer endpoints
+	allEndpoints map[string]map[string]struct{}
```

Comment:
where is the lock? 

1. Discovery accesses this from another goroutine
2. Updates for different channels take place in different goroutines